rosbag2_transport/src/rosbag2_transport/recorder.cpp

```diff
@@ -523,6 +533,23 @@ std::shared_ptr<rclcpp::GenericSubscription>
 RecorderImpl::create_subscription(
   const std::string & topic_name, const std::string & topic_type, const rclcpp::QoS & qos)
 {
+#ifdef _WIN32
+  if (!rmw_has_time_stamping_support_) {
```

Comment:
@jmachowinski We are going to use this variable only once and only in this place.  
Could you please replace it with the explicit condition? 
```
 if (std::string(rmw_get_implementation_identifier()).find("rmw_connextdds") != std::string::npos) {
```

Reply:
@jmachowinski Could you please address CI failures and my comments?
We need to merge this PR ASAP this week if we want it to be landed on the LTS Jazzy Jalisco release.

Reply:
ohh sorry, totally forgot about this PR, I was busy fixing race conditions in the executor for jazzy. I will try to address this tomorrow.

Reply:
Rebased, and changed as requested.